src/libsystemd-network/sd-dhcp6-lease.c

```diff
@@ -445,6 +445,37 @@ int sd_dhcp6_lease_get_fqdn(sd_dhcp6_lease *lease, const char **ret) {
         return 0;
 }
 
+int dhcp6_lease_set_captive_portal(sd_dhcp6_lease *lease, const uint8_t *optval, size_t optlen) {
+        _cleanup_free_ char *captive_portal = NULL;
+        int r;
+
+        assert(lease);
+        assert(optval || optlen == 0);
+
+        mfree(lease->captive_portal);
```

Comment:
Please do not free the previous result earlier, but use `free_and_replace()` at the end.